src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.netstandard.cs

```diff
@@ -101,6 +108,67 @@ public static unsafe AssemblyName GetAssemblyName(string assemblyFile)
             }
         }
 
+        private static void GetPEKind(PEReader peReader, out PortableExecutableKinds peKind, out ImageFileMachine machine)
+        {
+            PEHeaders peHeaders = peReader.PEHeaders;
+            PEMagic peMagic = peHeaders.PEHeader!.Magic;
+            Machine coffMachine = peHeaders.CoffHeader.Machine;
+            CorFlags corFlags = peHeaders.CorHeader!.Flags;
+
+            peKind = default;
+            if ((corFlags & CorFlags.ILOnly) != 0)
+                peKind |= PortableExecutableKinds.ILOnly;
+
+            if ((corFlags & CorFlags.Prefers32Bit) != 0)
+                peKind |= PortableExecutableKinds.Preferred32Bit;
+            else if ((corFlags & CorFlags.Requires32Bit) != 0)
+                peKind |= PortableExecutableKinds.Required32Bit;
+
+            if (peMagic == PEMagic.PE32Plus)
+                peKind |= PortableExecutableKinds.PE32Plus;
+
+            machine = (ImageFileMachine)coffMachine;
+        }
+
+        private static ProcessorArchitecture CalculateProcArchIndex(PortableExecutableKinds pek, ImageFileMachine ifm, AssemblyFlags flags)
+        {
+            if (((uint)flags & 0xF0) == 0x70)
```

Comment:
Any idea what this is trying to check for?

Reply:
There is one place that calls `0xF0` is `ProcArchMask`

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameFlags.cs#L13

I suspect `0x70` means `None`, but that is just a guess.

Reply:
Old NDP sources have this:

```cs
            // 0x70 specifies "reference assembly", a new concept in v4.0. For these, CLR wants to return None as arch so they
            // can be always loaded, regardless of process type. 
```

Reply:
I've checked a few reference assemblies. The COFF header specifies machine as `I386` and CorFlags are `ILOnly`, so this does look like needs to be handled in a special way. 
Otherwise the suggested simplified way will return `MSIL` while expected to return `None`